ql/src/java/org/apache/hadoop/hive/ql/exec/repl/incremental/IncrementalLoadTasksBuilder.java

```diff
@@ -314,13 +317,21 @@ private boolean shouldReplayEvent(FileStatus dir, DumpType dumpType, String dbNa
       return importTasks;
     }
 
+    boolean needCommitTx = updatedMetaDataTracker.isNeedCommitTxn();
+    // In migration flow, we should have only one table update per event.
+    if (needCommitTx && updatedMetaDataTracker.getUpdateMetaDataList().size() > 1) {
```

Comment:
I think it is not a possible error flow in any user operations and so it's dead code. Only if developer injected a bug, it can happen. So, it is better to assert it for developers to catch it.

Reply:
done